# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: df85fe3d7cb6
# Full Hash: df85fe3d7cb6a219c37e0a80d00f51e07783aec1
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 22 - Make nsIDragService::StartDragSession return the nsIDragSession r=rkraesig,win-reviewers,gstoll,geckoview-reviewers,m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D212298
# ==============================================================================

diff -r 384b025f56d3 -r df85fe3d7cb6 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/gtk/nsWindow.cpp	Thu Jul 04 07:48:12 2024 +0000
@@ -8718,9 +8718,8 @@
   if (!dragSession) {
     // This may be the start of an external drag session.
     nsIWidget* widget = window;
-    static_cast<nsDragSession*>(dragService->StartDragSession(widget));
     dragSession =
-      static_cast<nsDragSession*>(dragService->GetCurrentSession(window));
+        static_cast<nsDragSession*>(dragService->StartDragSession(widget));
   }
   NS_ENSURE_TRUE(dragSession, FALSE);
 